### ISSUE TITLE
unknown support for 'array-type'

### ISSUE THREAD

**Issue by ThomasdenH** [palantir/tslint#4085]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const a: unknown[] = [];
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "array-type": [true, "array-simple"],
  }
}
```

#### Actual behavior
TSLint recommends `Array<unknown>` instead of `unknown[]`.

#### Expected behavior
`unknown[]` is considered a simple type and thus be preferred over `Array<unknown>`.

**Comment by johnwiseheart** [comment]
We should definitely accept the `unknown` type as a simple type. Happy to review a PR for this.

**Comment by ThomasdenH** [comment]
Since `unknown` was added in Typescript 3.0.0, how to add this rule with backwards compatibility?
